src/utils.ts

```diff
@@ -0,0 +1,198 @@
+"use strict";
+
+import Table from "cli-table";
+import colors = require("colors/safe");
+import { FormatOptions } from "./index";
+
+interface TableOptions {
+    chars: Partial<Record<(
+        "top" |
+        "top-mid" |
+        "top-left" |
+        "top-right" |
+        "bottom" |
+        "bottom-mid" |
+        "bottom-left" |
+        "bottom-right" |
+        "left" |
+        "left-mid" |
+        "mid" |
+        "mid-mid" |
+        "right" |
+        "right-mid" |
+        "middle"
+    ), string>>;
+    truncate: string;
+    colors: boolean;
+    colWidths: number[];
+    colAligns: Array<"left" | "middle" | "right">;
+    style: Partial<{
+        "padding-left": number;
+        "padding-right": number;
+        head: string[];
+        border: string[];
+        compact: boolean;
+    }>;
+    head: string[];
+}
+
+/**
+ * @class
+ * @extends Table
+ */
+class StrippedTable extends Table {
+    constructor(StrippedTableConfig: Partial<TableOptions>) {
+        super(StrippedTableConfig);
+    }
+
+    public toString() {
+        return colors.strip(super.toString());
+    }
+}
+
+/**
+ * @typedef {Object} AssemblerConfig
+ * @property {boolean} oneTagPerLine Should tags be rendered one per line?
+ * @property {boolean} compact Should empty lines be added?
+ * @property {boolean} separateStepGroups Should step groups (when-then) be separated?
+ * @property {string} lineBreak The linebreak character(s)
+ * @property {string} indentation The indentation character(s)
+ */
+
+/** @type {AssemblerConfig} */
+const DEFAULT_OPTIONS = {
+    oneTagPerLine: false,
+    compact: false,
+    separateStepGroups: false,
+    lineBreak: "\n",
+    indentation: "  ",
+};
+
+const INDENTATION = "  ";
+
+/**
+ * Object to build a text file, from lines.
+ * @class
+ */
+export class Lines {
+    private options: Partial<FormatOptions>;
+    // tslint:disable-next-line:variable-name
+    private _lines: string[];
+
+    get lines(): string[] {
+        return this._lines;
+    }
+
+    /**
+     * @constructor
+     * @param {AssemblerConfig} [options]
+     */
+    constructor(options: Partial<FormatOptions>) {
+        /** @member {boolean} */
+        this.options = config(options);
+        /**
+         * @type {Array<string>}
+         * @private
+         */
+        this._lines = [];
+    }
+
+    /**
+     * Adds a new line with the given text.
+     * If the text is not passed,
+     * it will be an empty line
+     * @param {...string|*} [texts]
+     */
+    public add(...texts: string[]): void {
```

Comment:
I'd suggest returning the current instance from this and the other functions of a `Lines` instance where it is `void` right now. In this way, you could chain the operations and make the code shorter and more readable, e.g.
```typescript
return lines(options).add(formatTableRows(dataTable.rows)).toString();
```